packages/frontend-v2/pages/home.tsx

```diff
@@ -175,6 +206,43 @@ const HomePage: NextPage = () => {
   );
 };
 
+interface DeletePlanButtonProps {
+  onOpen: () => void;
+  onClose: (isDeleted: boolean) => void;
+  isOpen: boolean;
+  planName: string;
+  planId: number;
+}
+
+const DeletePlanButton: React.FC<DeletePlanButtonProps> = ({
+  onOpen,
```

Comment:
I think we should move this button (and the disclosure hooks and logics) to the `DeletePlan` component, so that it encapsulates the modal logic better. The new modal will look something like
```Typescript
type DeletePlanModalProps = {
   planName: string;
   planId: number;
}

const DeletePlanModal: React.FC<DeletePlanModalProps>({
   planName,
   planId,
}) => {
   const { onOpen, onClose, isOpen } = useDisclosure();
   
   return (
        <>
             <Button/>
             <Modal>
                    ...
             <Modal/>
        </>
   )
}
```

This will remove the similar type (`DeletePlanButtonProps` and `DeletePlanModalProps `), remove the `useDisclosure` hook from cluttering the `Home` page (and also remove the conflicting name from the modals), and the `onOpen`, `onClose` and `isOpen` parameters from the modal component.

Reply:
agree that it makes more sense to have the delete plan button live in the components directory and be imported into home.tsx-- it'd be nice to do something similar for the add modal button as well! :)

Reply:
good point, thank guys!